commit 4cfb7c89f27b9b28a3324751a9c85a9351544a72
Author: Trent Mick <trentm@gmail.com>
Date:   2019-07-25T10:22:00-07:00 (2 months ago)
    
    TOOLS-2298 check-copyright fails for non-master Jenkins builds
    Reviewed by: Tim Foster <tim.foster@joyent.com>
    Approved by: Tim Foster <tim.foster@joyent.com>

diff --git a/tools/check-copyright b/tools/check-copyright
index b260b93..9fa6130 100755
--- a/tools/check-copyright
+++ b/tools/check-copyright
@@ -145,7 +145,13 @@ done
 shift $((OPTIND - 1))
 
 # Gather the files to possibly check.
-committedFiles=$(git diff --name-only master..)
+committedFiles=
+if git branch -a | grep '^  remotes/origin/master$' >/dev/null 2>/dev/null; then
+    # The intent for committed files is to compare against the ultimate
+    # merge branch target. We assume that is "origin/master", the common case
+    # for Joyent eng repos.
+    committedFiles=$(git diff --name-only origin/master..)
+fi
 stagedFiles=$(git diff --name-only --staged)
 unstagedFiles=$(git diff --name-only)
 uniqFiles=$(echo "
